### PR TITLE
Fix advanced AR tests

### PR DESCRIPTION
**Pull request by RowanDW**

### CODE REVIEW

File: spec/models/customer_spec.rb [hunk 1]
```diff
@@ -2,47 +2,46 @@
 
 RSpec.describe Customer, type: :model do
   before :each do
-    @joey = Customer.create!(first_name: "Joey", last_name: "Ondricka", created_at: "2012-03-27 14:54:09 UTC", updated_at: "2012-03-27 14:54:09 UTC")
-    @cecelia = Customer.create!(first_name: "Cecelia", last_name: "Osinski", created_at: "2012-03-27 14:54:10 UTC", updated_at: "2012-03-27 14:54:10 UTC")
-    @mariah = Customer.create!(first_name: "Mariah", last_name: "Toy", created_at: "2012-03-27 14:54:10 UTC", updated_at: "2012-03-27 14:54:10 UTC")
-    @donna = Customer.create!(first_name: "Donna", last_name: "Dumdooko", created_at: "2012-03-27 14:54:10 UTC", updated_at: "2012-03-27 14:54:10 UTC")
-    @christ = Customer.create!(first_name: "Christ", last_name: "Hough", created_at: "2012-03-27 14:54:10 UTC", updated_at: "2012-03-27 14:54:10 UTC")
-    @amy = Customer.create!(first_name: "Amy", last_name: "Hollerway", created_at: "2012-03-27 14:54:10 UTC", updated_at: "2012-03-27 14:54:10 UTC")
+    @merchant = create(:merchant)
```
Comment (ozzman84) [inline spec/models/customer_spec.rb:5]: Great way to DRY up the code.

File: spec/models/customer_spec.rb [hunk 2]
```diff
@@ -58,21 +57,20 @@
...
     end
 
     describe '#top_merchant_customers' do
       it "returns the customers with the highest transaciton count for the merchant" do
-        Transaction.create!(invoice_id: @invoice_4.id, credit_card_number: "4203696133194408", result: "success", created_at: "2012-03-27 14:54:10 UTC", updated_at: "2012-03-27 14:54:10 UTC")
-        Transaction.create!(invoice_id: @invoice_4.id, credit_card_number: "4203696133194408", result: "success", created_at: "2012-03-27 14:54:10 UTC", updated_at: "2012-03-27 14:54:10 UTC")
-        Transaction.create!(invoice_id: @invoice_2.id, credit_card_number: "4203696133194408", result: "success", created_at: "2012-03-27 14:54:10 UTC", updated_at: "2012-03-27 14:54:10 UTC")
-        expect(Customer.top_merchant_customers(@zara.id)).to eq([@donna, @cecelia, @joey])
+        expect(Customer.top_merchant_customers(@merchant)).to eq([@customers[5], @customers[1], @customers[4], @customers[3], @customers[2]])
+        expect(Customer.top_merchant_customers(@merchant2)).to eq([@customer2])
```
Comment (ozzman84) [inline spec/models/customer_spec.rb:73]: Stong test that will account for edge cases!

**Review by ozzman84 — APPROVED** [review]
